### ISSUE TITLE
Add class identifier to json2typescript class decorator

### ISSUE THREAD

**Issue by andreas-aeschlimann** [dhlab-basel/Salsah#278]
Starting from json2typescript v1.0.6, it is **strongly** recommended to use a class identifier in the class decorator. Example:

```
@JsonObject("User")
export class User {
    ...
}
```

The reason: If you minimize large projects, class names may become a single letter. This makes it impossible for json2typescript to show a good debug message. It is also possible that some mapping classes collide.

See also https://github.com/dhlab-basel/json2typescript/issues/66 or the readme on https://github.com/dhlab-basel/json2typescript/

**Comment by subotic** [comment]
As far as I understand it, anything `json2typescript` related should be removed from this project, when the refactoring with Knora-libraries is done. @flavens did I understand this correctly?

**Comment by kilchenmann** [comment]
Yes, we‘re using the json2typescript in the knore core module, which has his own repository. I will create an issue there.
But with the usage of the core module, the json2typescript is still a part of salsah. So, thanks @andreas-aeschlimann for the update and information

**Comment by andreas-aeschlimann** [comment]
Ah, I didn't realize that. Thanks for the heads up @subotic @kilchenmann.

**Comment by kilchenmann** [comment]
Easy

**Comment by flavens** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [@knora/core refact some classes with @JsonObject/@Json/property (core/declarations/api/v2)](https://trello.com/c/MZ8enVck/15-knora-core-refact-some-classes-with-jsonobject-json-property-core-declarations-api-v2)
